File: 30-days-of-code/java/Day9.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Day9 {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int result = factorial(n);
        System.out.println(result);
    }

    private static int factorial(int num) {
        if (num <= 1) return num;
        return num * factorial(num - 1);
    }
}

File: 30-days-of-code/go/Day9.go
package main

import "fmt"

func factorial(n int) int {
	if n <= 1 {
		return n
	}
	return n * factorial(n-1)
}

func main() {
	var n int
	fmt.Scanf("%d", &n)
	result := factorial(n)

	fmt.Println(result)
}
